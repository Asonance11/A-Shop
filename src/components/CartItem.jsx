import React from 'react';

const CartItem = ({ item, cart, setCart }) => {
	return (
		<section>
			<img src={item.image} alt={item.alt} />
			<p>{item.photographer}</p>
			<p>{item.price}</p>
			<p>{item.quantity}</p>
			<button type="button">Remove from Cart</button>
		</section>
	);
};

export default CartItem;
