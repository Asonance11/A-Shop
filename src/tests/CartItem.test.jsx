import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, it, vi } from 'vitest';
import CartItem from '../components/CartItem';

describe('CartItem component', () => {
	it('renders correctly', async () => {
		const setCart = vi.fn();
		const anItem = {
			image: 'www.awebsite.com/something.jpeg',
			alt: 'An image',
			photographer: 'me',
			price: 10000,
			imageSize: '3000 x 3000',
		};

		const { container } = render(
			<CartItem item={anItem} cart={[]} setCart={setCart} />,
			{
				wrapper: MemoryRouter,
			}
		);
		expect(container).toMatchSnapshot();
	});

	it('calls setCart when "remove from cart button" is clicked', async () => {
		const setCart = vi.fn();
		const anItem = {
			image: 'www.awebsite.com/something.jpeg',
			alt: 'An image',
			photographer: 'me',
			price: 10000,
			imageSize: '3000 x 3000',
		};
		const user = userEvent.setup();

		render(<CartItem item={anItem} cart={[]} setCart={setCart} />);

		const removeButton = screen.getByRole('button', {
			name: 'Remove from Cart',
		});

		await user.click(removeButton);

		expect(setCart).toHaveBeenCalled();
	});
});
