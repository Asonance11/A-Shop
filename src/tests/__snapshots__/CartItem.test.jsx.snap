// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartItem component > renders correctly 1`] = `
<div>
  <section
    class="flex items-center gap-4 border text-white font-sans"
  >
    <div>
      <img
        alt="An image"
        class=" h-[22vh] object-cover"
        src="www.awebsite.com/something.jpeg"
      />
    </div>
    <div
      class="w-full"
    >
      <p
        class="capitalize"
      >
        me
      </p>
      <p>
        $
        10000
      </p>
      <p>
        Quantity: 
      </p>
      <button
        class="py-2 px-4 bg-white text-darkGrey rounded-md mt-2"
        type="button"
      >
        Remove from Cart
      </button>
    </div>
  </section>
</div>
`;
